src/clj/puppetlabs/http/client/async.clj

```diff
@@ -293,14 +294,36 @@
         :else
           (DefaultRedirectStrategy.))))
 
+(schema/defn request-config :- RequestConfig
```

Comment:
I agree that it would be worthwhile to eliminate some of this duplication between the Clojure and Java implementations, but I think it should be done as a broader effort around the request setup.  I filed a separate ticket to cover this idea - https://tickets.puppetlabs.com/browse/TK-179.